hibernate-core/src/main/java/org/hibernate/loader/BatchFetchStyle.java

```diff
@@ -39,7 +39,11 @@
 	 * Dynamically builds its SQL based on the actual number of available ids.  Does still limit to the batch-size
 	 * defined on the entity/collection
 	 */
-	DYNAMIC;
+	DYNAMIC,
+	/**
+	 * Use the ANY SQL operator and pass the identifier as an ARRAY.
+	 */
+	ANY_ARRAY;
```

Comment:
The name is database-specific. It could be supported by other DBs in the future as well. It just says what it does: 

- an ANY clause is used
- and the ANY clause takes an ARRAY

Reply:
`ANY` is PostgreSQL specific. It can have a totally different meaning depending on the database. See for instance https://docs.microsoft.com/en-us/sql/t-sql/language-elements/some-any-transact-sql?view=sql-server-2017 .

I would go with `ARRAY`. It seems to be generic enough and even if the database is not exactly using the same array notion, it makes sense to consider them arrays anyway.

Reply:
Nope. ANY means the same thing on any DBs. The only thing that PostgreSQL does differently is that it allows ARRAYs to be passed instead of a TABLE result (e.g. subselect). It might be that Oracle supports this in a similar fashion, but I need to test it.